test/src/main/java/org/apache/accumulo/harness/AccumuloClusterHarness.java

```diff
@@ -217,6 +219,17 @@ public void setupCluster() throws Exception {
     }
   }
 
+  public void checkConnection() {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      // Use prefix so cleanupTables deletes it
+      String tablePrefix = this.getClass().getSimpleName() + "_";
+      client.tableOperations().create(tablePrefix);
```

Comment:
Could use an operation that doesn't affect any tables or create anything, like ```list()```

Reply:
Yeah, list is a good one that I forgot to try. I liked the instance operations for the same reasoning. I can push your suggestion tomorrow if thats more preferred.

Reply:
Why is just getting the client not sufficient?  

Other options could be InstanceOperations.getSystemConfiguration or getSiteConfiguration()

Another possibility would be to see if you can scan either the root or the metadata table, and if you restricted the colFam to file or ~tab:~pr - that's only going to be a few entries - and that's a fairly robust way of know that a lot is up and ready for operations.

Reply:
> Why is just getting the client not sufficient?

I'm not sure getting the client requires any communication with a server. I don't think it does... it's all client side. We'd need something that actually interacts with the server.

Reply:
> I'm not sure getting the client requires any communication with a serve

That is correct.